lib/samuel/checks/has_comments.ex

```diff
@@ -33,16 +33,19 @@ defmodule Samuel.Checks.HasComments do
   end
 
   def requirements do
-    ~w(comments)a
+    ~w(commit_comments pr_comments)a
   end
 
+  defp all_comments(data) do
+    data.commit_comments ++ data.pr_comments
+  end
 
   # Returns the number of comments made by someone who is not the Pull Request
   # author or Samuel.
   defp other_user_comments(data) do
     users_that_dont_count = users_that_dont_count(data.event)
```

Comment:
In Elixir functions and variable live in two different namespaces, we don't have objects with mutable state you can use for memoization, and it's not Haskell with lazy evaluation, so we want to be explicit about where and when we do computation.